### PR TITLE
Improve tether based on feedback

### PR DESCRIPTION
**Pull request by Fueredoriku**
- Plugging in now works by throwing the plug to a surface instead of being locked in a long animation
- Tether starts unplugged to make players in charge of initial plug position
- Wire being pulled back through all collision points
- More visual feedback in terms of the plug being damageable
- New soundFXs for better feedback on tether actions

TODO in another PR:
- Pluged in / not pluged on HUD instead of normal ammo HUD

### CODE REVIEW

**Review by toberge — CHANGES_REQUESTED** [review]
Works much better now, and I love the animations for throwing and retracting 👍 

However, there is an issue that I can't look past here:
The cable blends all too well in with the ground. May I suggest making it blue instead? 😅 or just a different color that has better contrast with the ground
image

Additionally, I notice that decals stick to the plug, which looks quite odd:
image

Will look through the code and give you some feedback on that too, but the cable color is really crucial 😅

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs [hunk 1]
```diff
@@ -73,48 +91,150 @@ private void PullingWire()
...
+            rope.enabled = true;
+            plugAnchor.gameObject.SetActive(true);
+            AnimateFullPullBack();
+        }
+    }
+
+    public void PlayThrowAudio()
+    {
+        audioSource.clip = plugThrow;
+        audioSource.Play();
```
Comment (toberge) [inline Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs]: Please use an AudioGroup for this, as this current version will not be spatialized whatsoever

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs [hunk 2]
```diff
@@ -73,48 +91,150 @@ private void PullingWire()
...
-        rope.enabled = false;
-        gunController.stats.Ammo = 0;
-        plugAnchor.gameObject.SetActive(false);
-        audioSource.Play();
+        if (isWired == false)
+            return;
+        RemoveRope();
     }
 
     private void CheckForWirePlanting(GunStats stats)
```
Comment (toberge) [inline Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs]: Perhaps rename to `Throw` or something?

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs [hunk 3]
```diff
@@ -73,48 +91,150 @@ private void PullingWire()
...
+                    gunController.stats.Ammo = gunController.stats.MagazineSize;
+                });
+            rope.ResetRope(plugAnchor.WireOrigin);
+            plugAnchor.gameObject.SetActive(true);
+        }
+        else
+        {
+            plugAnchor.SetPulseStrength(0);
+            plugAnchor.transform.position = gunController.Player.transform.position;
+            plugAnchor.transform.forward = gunController.Player.inputManager.transform.forward;
```
Comment (toberge) [inline Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs:171]: FYI, there's a brief moment where the plug is visible on the player, I think it looks a lil weird

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs [hunk 4]
```diff
@@ -73,48 +91,150 @@ private void PullingWire()
...
+            {
+                    rope.enabled = false;
+                    isThrowing = false;
+                    plugAnchor.gameObject.SetActive(false);
+                    canThrow = true;
+                    rope.CollisionCheckActive = true;
+            });
+    }
+
+    public void SetThrowValue(float value)
```
Comment (toberge) [inline Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs]: SetThrowValue? What does this even do? The value is unused and the method name is a lil confusing considering it just resets the rope 🤔

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs [hunk 5]
```diff
@@ -73,48 +91,150 @@ private void PullingWire()
...
+        audioSource.clip = plugThrow;
+        audioSource.Play();
+    }
+
+    private void AnimateFullPullBack()
+    {
+        LeanTween.value(gameObject, SetThrowValue, 0f, ropeLength, 0.02f * ropeLength)
+            .setEaseOutElastic();
+
+        plugAnchor.gameObject.LeanMove(gunController.Player.inputManager.transform.position + gunController.Player.inputManager.transform.forward * ropeLength, 0.02f * ropeLength)
```
Comment (toberge) [inline Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs:187]: This may look a little odd if the player moves, as far as I can tell?

Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs:187]: The cause was from another place, have fixed it now 👍

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs (hunk 4 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs]: Renamed

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs (hunk 3 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs:171]: Have tried hiding as best as I can now

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs (hunk 2 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs]: Done 👍

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs (hunk 1 above)
Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs]: Done 👍

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs (hunk 3 above)
Comment (toberge) [reply Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs:171]: Seems to have worked well 👍

File: Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs (hunk 5 above)
Comment (toberge) [reply Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs:187]: Seems that way 👍

**Review by toberge — APPROVED** [review]
Much better contrast with the ground now 👍

**Pull request merged**